### ISSUE TITLE
force_index only works on MySQL 5.7

### ISSUE THREAD

**Issue by Gwildor** [adamchainz/django-mysql#305]
Summary: `force_index` only works on MySQL 5.7.

Django Version: 1.7.10
Database and version used: MySQL 5.6
Version: Django-MySQL 1.0.7

Hi there. I've been trying to get your plugin to work because I needed the `force_index` feature, but unfortunately, it only seems to work on MySQL 5.7 because your implementation depends on the Query Rewrite plugin being installed and enabled. This was not documented, unfortunately. Is there a specific reason you are using the Query Rewrite plugin, instead of simply adding a `FORCE INDEX` clause on the query, which works fine?

**Comment by adamchainz** [comment]
It doesn't use Query Rewrite at all and is tested on MySQL 5.5+ and MariaDB 10.0+. What's your exact query set?

**Comment by Gwildor** [comment]
The query it generated for me is something like this:

```sql
SELECT * FROM `activities_activity` INNER JOIN `users_user` ON ( `activities_activity`.`user_id` = `users_user`.`id` ) WHERE (/*QueryRewrite':index=`activities_activity` FORCE `activities_activity_timestamp`*/1) ORDER BY `activities_activity`.`timestamp` DESC
```

And the queryset:

```python
models.Activity.objects.select_related('user').force_index('activities_activity_timestamp')
```

Removing the `select_related` doesn't change anything.

**Comment by adamchainz** [comment]
I think you don't have `DJANGO_MYSQL_REWRITE_QUERIES = True` in your settings, so the query rewrite hint gets written by Django-MySQL but it hasn't patched Django's `cursor.execute` so it doesn't rewrite it. See the introduction at: http://django-mysql.readthedocs.io/en/latest/queryset_extensions.html#query-hints

I guess `force_index` etc. could check that the setting is set and error if this is not the case

**Comment by adamchainz** [comment]
Will fix the lack of error message in #306. If this was your problem please close the issue. Thx.

**Comment by Gwildor** [comment]
Yes, enabling this setting seems to have fixed this, thank you. I see now that it *is* documented, sorry about that. An error message when calling one of those methods without the setting enabled would be appreciated. Thanks for your responses!
